src/main/java/edu/wpi/grip/ui/pipeline/input/NumberSpinnerInputSocketView.java

```diff
@@ -0,0 +1,66 @@
+package edu.wpi.grip.ui.pipeline.input;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.events.SocketChangedEvent;
+import javafx.beans.InvalidationListener;
+import javafx.scene.control.Spinner;
+import javafx.scene.control.SpinnerValueFactory;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * An {@link InputSocketView} that lets the user set the value of a number with a {@link Spinner}
+ */
+public class NumberSpinnerInputSocketView extends InputSocketView<Number> {
+
+    private final static Object[] DEFAULT_DOMAIN = new Object[]{Double.MIN_VALUE, Double.MAX_VALUE};
+
+    private final SpinnerValueFactory<Double> valueFactory;
+    private final InvalidationListener updateSocketFromSpinner;
+
+    /**
+     * @param socket An <code>InputSocket</code> with a domain containing two <code>Number</code>s (the min and max
+     *               slider values), or no domain at all.
+     */
+    public NumberSpinnerInputSocketView(EventBus eventBus, InputSocket<Number> socket) {
+        super(eventBus, socket);
+
+
+        Object[] domain = socket.getSocketHint().getDomain();
+        if (domain == null) {
+            domain = DEFAULT_DOMAIN;
+        }
+
+        checkArgument(domain.length == 2 && domain[0] instanceof Number && domain[1] instanceof Number,
+                "Spinners must have a domain with two numbers (min and max)");
```

Comment:
What I'm trying to avoid is having operations with invalid socket hints in our release application.
Perhaps a better way of handling this problem is to have a UI test that adds every single one of our operations to the pipeline and ensures that no errors are thrown.

If you think this is a better plan I'll do that instead.